cmd/generate.go

```diff
@@ -1,21 +1,118 @@
 package cmd
 
 import (
+	"bytes"
+	"errors"
 	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
 
+	k8yaml "k8s.io/apimachinery/pkg/util/yaml"
+
+	kube "github.com/IBM/argocd-vault-plugin/pkg/kube"
 	"github.com/spf13/cobra"
 )
 
-// NewGenerateCommand Initializes the generate command
+// NewGenerateCommand initializes the generate command
 func NewGenerateCommand() *cobra.Command {
 	var command = &cobra.Command{
-		Use:   "generate",
-		Short: "Generate YAML files from template with Vault values",
+		Use:   "generate <path>",
+		Short: "Generate manifests from templates with Vault values",
 		RunE: func(cmd *cobra.Command, args []string) error {
-			fmt.Fprint(cmd.OutOrStdout(), "Generate YAML")
+
+			path := args[0]
+			files := listYamlFiles(path)
+			if len(files) < 1 {
+				return fmt.Errorf("no YAML files were found in %s", path)
+			}
+			manifests, errs := readFilesAsManifests(files)
+			if len(errs) != 0 {
+
+				// TODO: handle multiple errors nicely
+				return fmt.Errorf("could not read YAML files: %s", errs)
+			}
+
+			var resource kube.Template
+			for _, manifest := range manifests {
```

Comment:
I would say lets pull this switch out and have a method that returns a `Template` interface